packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -319,9 +319,13 @@ void debugDumpLayerTree() {
 
 /// Prints a textual representation of the entire semantics tree.
 /// This will only work if there is a semantics client attached.
-/// Otherwise, the tree is empty and this will print "null".
-void debugDumpSemanticsTree() {
-  debugPrint(RendererBinding.instance?.renderView?.debugSemantics?.toStringDeep() ?? 'Semantics not collected.');
+/// Otherwise, a notice that no semantics are available will be printed.
+///
+/// By default, children of a [SemanticsNode] are printed in traversal order.
+/// This can be changed to inverse hit test order by setting
+/// [childrenInInverseHitTestOrder] to `true`.
+void debugDumpSemanticsTree({ bool childrenInInverseHitTestOrder: false }) {
```

Comment:
what is this argument for?

Reply:
It's explained in the doc comment right above: It controls whether the children of a SemanticsNode are dumped in traversal order or in inverse hit test order. Both can be useful depending on what you're trying to debug. If you're debugging the semantic information (e.g. why does a node have a certain action or label) its much easier and far less confusing if the tree is printed in traversal order. However, for the (rare) case that you need to debug hit testing on semantics nodes, you want to dump the other order. This param gives you the option.

I mainly call this from test code for debugging purposes.

Reply:
I don't think we should do this. It's going to make porting SemanticsNode to the DiagnosticableTree logic much harder.

Reply:
Let's make this an enum (defined in debug.dart), DebugSemanticsDumpOrder or some such, rather than a boolean.

Reply:
also let's make it a positional argument, not a named argument.